crates/sui-benchmark/src/util.rs

```diff
@@ -20,3 +40,184 @@ pub fn get_ed25519_keypair_from_keystore(
+                return (updated, minted);
+            }
+            Err(err) => {
+                error!("Error while transferring sui: {:?}", err);
+                tokio::time::sleep(Duration::from_secs(3)).await;
+            }
+        }
+    }
+    panic!("Failed to finish transfer_sui_for_testing");
+}
+pub fn make_split_coin_tx(
+    framework: ObjectRef,
+    sender: SuiAddress,
+    coin: &Object,
+    split_amounts: Vec<u64>,
+    gas: ObjectRef,
+    keypair: &AccountKeyPair,
+) -> VerifiedTransaction {
+    let split_coin = TransactionData::new_move_call(
+        sender,
+        framework,
+        coin::PAY_MODULE_NAME.to_owned(),
+        coin::PAY_SPLIT_VEC_FUNC_NAME.to_owned(),
+        vec![],
+        gas,
+        vec![
+            CallArg::Object(ObjectArg::ImmOrOwnedObject(coin.compute_object_reference())),
+            CallArg::Pure(bcs::to_bytes(&split_amounts).unwrap()),
+        ],
+        MAX_GAS_FOR_TESTING,
+    );
+    let verified_tx = to_sender_signed_transaction(split_coin, keypair);
+    verified_tx
+}
+
+pub fn make_pay_tx(
+    input_coins: Vec<ObjectRef>,
+    sender: SuiAddress,
+    addresses: Vec<SuiAddress>,
+    split_amounts: Vec<u64>,
+    gas: ObjectRef,
+    keypair: &AccountKeyPair,
+) -> VerifiedTransaction {
+    let pay = TransactionData::new_pay(sender, input_coins, addresses, split_amounts, gas, MAX_GAS);
+    let verified_tx = to_sender_signed_transaction(pay, keypair);
+    verified_tx
+}
+
+pub async fn split_coin_and_pay(
+    proxy: Arc<dyn ValidatorProxy + Send + Sync>,
+    coin: &Object,
+    coin_configs: Vec<GasCoinConfig>,
+    gas: Gas,
+) -> Result<UpdatedAndNewlyMintedGasCoins> {
+    // split one primary gas token into amounts
+    // send those amounts to addresses
+    let sender = coin.owner.get_owner_address()?;
+    let framework = proxy
+        .get_object(SUI_FRAMEWORK_OBJECT_ID)
+        .await?
+        .compute_object_reference();
+    let split_amounts: Vec<u64> = coin_configs.iter().map(|c| c.amount).collect();
+    let verified_tx = make_split_coin_tx(
+        framework,
+        sender,
+        coin,
+        split_amounts.clone(),
+        gas.0,
+        &gas.2,
+    );
+    let (_, effects) = proxy.execute_transaction(verified_tx.into()).await?;
+    let updated_gas = effects
+        .mutated()
+        .into_iter()
+        .find(|(k, _)| k.0 == gas.0 .0)
+        .ok_or("Input gas missing in the effects")
+        .map_err(Error::msg)?;
+    let created_coins: Vec<ObjectRef> = effects.created().into_iter().map(|c| c.0).collect();
+    assert_eq!(created_coins.len(), split_amounts.len());
+    let created_coin_ids: Vec<ObjectID> = created_coins.iter().map(|c| c.0).collect();
+    let recipient_addresses: Vec<SuiAddress> = coin_configs.iter().map(|x| x.address).collect();
+    let verified_tx = make_pay_tx(
+        created_coins,
+        sender,
+        recipient_addresses,
+        split_amounts,
+        updated_gas.0,
+        &gas.2,
```

Comment:
I think you should just use pay_sui.

Reply:
It seems like the only difference b/w `pay` (which is being used here) and `pay_sui` is that you don't need an extra gas coin to pay for the transaction itself. You still need to split the coin into multiple coins which can be transferred to the recipients - so in the end you still need two transactions? This is my understanding but I could be wrong.

Reply:
No you don't need to split them. pay_sui splits the coins out for you.

Reply:
Adding TODO for it, will take it up as a follow up. Thanks